variables field on runs api (#2007)
hashicorp/terraform-website: content/source/docs/cloud/api/run.html.md

Creating a run requires permission to queue plans for the specified workspace. ([More about permissions.](/docs/cloud/users-teams-organizations/permissions.html))

> When creating a run, you may optionally provide a list of variable objects containing key and value attributes. These values claim a high priority relative to variables defined at the workspace level. See [Variables Precedence](https://www.terraform.io/docs/cloud/workspaces/variables.html#precedence) section for more information. All values must be expressed as an HCL literal in the same syntax you would use when writing terraform code. See [Terraform types](https://www.terraform.io/docs/language/expressions/types.html#types) for more details.

Reviewer: ```suggestion
When creating a run, you may optionally provide a list of variable objects containing key and value attributes. These values apply to that run specifically and take precedence over variables with the same key applied to the workspace(e.g., variable sets). Refer to [Variable Precedence](https://www.terraform.io/docs/cloud/workspaces/variables.html#precedence) for more information. All values must be expressed as an HCL literal in the same syntax you would use when writing Terraform code. Refer to [Types](https://www.terraform.io/docs/language/expressions/types.html#types) in the Terraform documentation for more details.
```

Properties without a default value are required.

Key path                                           | Type                | Default                                            | Description
---------------------------------------------------|---------------------|----------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

> `data.attributes.auto-apply`                       | bool                | (The current Auto Apply option of the workspace)   | Whether to automatically apply changes when a Terraform plan is successful. Defaults to the [workspace Auto Apply](https://www.terraform.io/docs/cloud/workspaces/settings.html#auto-apply-and-manual-apply) setting.

Reviewer: ```suggestion
`data.attributes.auto-apply`                       | bool                | The current Auto Apply setting in the workspace  | Whether to automatically apply changes when a Terraform plan is successful. Defaults to the [Auto Apply](/docs/cloud/workspaces/settings.html#auto-apply-and-manual-apply) workspace setting.
```

> `data.attributes.is-destroy`                       | bool                | false                                              | Specifies if this plan is a destroy plan, which will destroy all provisioned resources. Mutually exclusive with `refresh-only`.

Reviewer: ```suggestion
`data.attributes.is-destroy`                       | bool                | false                                              | Specifies whether this plan is a destroy plan that will destroy all provisioned resources. Mutually exclusive with `refresh-only`.
```

`data.attributes.message`                          | string              | "Queued manually via the Terraform Enterprise API" | Specifies the message to be associated with this run.
`data.attributes.refresh`                          | bool                | true                                               | Specifies whether or not to refresh the state before a plan.

> `data.attributes.refresh-only`                     | bool                | false                                              | Whether this run should use the refresh-only plan mode, which will refresh the state without modifying any resources. Mutually exclusive with `is-destroy`.

Reviewer: ```suggestion
`data.attributes.refresh-only`                     | bool                | false                                              | Specifies whether this run should use the refresh-only plan mode, which will refresh the state without modifying any resources. Mutually exclusive with `is-destroy`.
```

`data.attributes.replace-addrs`                    | array[string]       | (nothing)                                          | Specifies an optional list of resource addresses to be passed to the `-replace` flag.
`data.attributes.target-addrs`                     | array[string]       | (nothing)                                          | Specifies an optional list of resource addresses to be passed to the `-target` flag.

> `data.attributes.variables`                        | array[{key, value}] | empty                                              | Specifies an optional list of run-specific variable values. See [Run-Specific Variables](https://www.terraform.io/docs/cloud/workspaces/managing-variables.html#run-specific-variables) in the Terraform Cloud workspaces documentation for more information.

Reviewer: ```suggestion
`data.attributes.variables`                        | array[{key, value}] | empty                                              | Specifies an optional list of run-specific variable values. Refer to [Run-Specific Variables](https://www.terraform.io/docs/cloud/workspaces/managing-variables.html#run-specific-variables) in the Terraform Cloud documentation for more information.
```

`data.relationships.workspace.data.id`             | string              | (nothing)                                          | Specifies the workspace ID where the run will be executed.

> `data.relationships.configuration-version.data.id` | string              | (nothing)                                          | Specifies the configuration version to use for this run. If the `configuration-version` object is omitted, the run will be created using the workspace's latest configuration version.

Reviewer: Is (nothing) a convention we usually use? If so I accept it. If not, I wonder is there a convention we typically use for no default. Or, should we just leave that cell blank?
